src/main/app/utils.ts

```diff
@@ -38,8 +38,12 @@ const log = new Logger('App.Utils');
 
 export function openDeepLink(deeplinkingUrl: string) {
     try {
-        MainWindow.show();
-        ViewManager.handleDeepLink(deeplinkingUrl);
+        if (MainWindow.get()) {
+            MainWindow.show();
+            ViewManager.handleDeepLink(deeplinkingUrl);
+        } else {
+            MainWindow.on(MAIN_WINDOW_CREATED, () => ViewManager.handleDeepLink(deeplinkingUrl));
```

Comment:
Are there any race conditions at play? (Could `MainWindow` get created between our `get` and `on` registration?

Reply:
Well I went to go double-check, I found another instance where we're calling `show()` before `app.ready()` which is really not good.

However, other callers of this are also waiting for the `ready` event, so as long as we're calling it after that there shouldn't be any race conditions as that's when we expect the window to be registered.

The `ready` event docs if you're interested: https://www.electronjs.org/docs/latest/api/app#event-ready